core/base/src/TApplication.cxx

```diff
@@ -628,20 +630,453 @@ TApplication::EExitOnException TApplication::ExitOnException(TApplication::EExit
    return old;
 }
 
+/////////////////////////////////////////////////////////////////////////////////
+/// The function generates and executes a command that loads the Doxygen URL in
+/// a browser. It works for Mac, Windows and Linux. In the case of Linux, the
+/// function also checks if the DISPLAY is set. If it isn't, a warning message
+/// and the URL will be displayed on the terminal.
+///
+/// \param[in] url <web page to be displayed in a browser
+
+void TApplication::OpenInBrowser(const TString &url)
+{
+   // We check what operating system the user has.
+      #ifdef R_MACOSX
+      // Command for opening a browser on Mac.
+      TString cMac("open ");
+      // We generate the full command and execute it.
+      cMac.Append(url);
+      gSystem->Exec(cMac);
+       #elif R__WIN32
+      // Command for opening a browser on Windows.
+      TString cWindows("start ");
+      cWindows.Append(url);
+      gSystem->Exec(cWindows);
+       #else
+      // Command for opening a browser in Linux.
+      TString cLinux("xdg-open ");
+      // For Linux we check if the DISPLAY is set.
+      //TString displayHandle(gSystem->Getenv("DISPLAY"));
+      //if ((displayHandle.Sizeof()) > 1) {
+      if (gSystem->Getenv("DISPLAY")) {
+         // If the DISPLAY is set it will open the browser.
+         cLinux.Append(url);
+         gSystem->Exec(cLinux);
+      } else {
+         // Else the user will have a warning and the URL in the terminal.
+         Warning("OpenInBrowser", "The DISPLAY is not set!");
+         std::cout << url.Data() << std::endl;
+      }
+      #endif
+}
+
+enum EUrl {kURLforClass, kURLforNameSpace};
+namespace {
 ////////////////////////////////////////////////////////////////////////////////
-/// Print help on interpreter.
+/// The function generates a URL address for class or namespace (name).
+/// This is the URL to the online reference guide, generated by Doxygen.
+/// With the enumeration "EUrl" we pick which case we need - the one for
+/// class (kURLforClass) or the one for namespace (kURLforNameSpace).
+///
+/// \param[in] name <the name of the class or the namespace
+/// \param[in] type <the enumerator for class or namespace
 
-void TApplication::Help(const char *line)
+static TString UrlGenerator(const TString &name, EUrl type)
 {
-   gInterpreter->ProcessLine(line);
+   TString url("");
+   if (type == kURLforClass){
+      url.Append("https://root.cern.ch/doc/master/class");
```

Comment:
`url = ` is more idiomatic (`Append` makes me wonder "what do we append to" and I need to check; `=` makes clear that it overwrites anything that might have been inside the string, so previous state is irrelevant.)